src/hooks/useGrantPermissions.ts

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from 'react'
```

Comment:
Good idea to create a hook for this, but i think it really makes sense to use it everywhere, so please adjust CameraOverlay.tsx, so it uses this hook.

Reply:
i adjusted the hook to make it work for `CameraOverlay` and added in there